### PR TITLE
Add 'Expiry Time Considerations' section

### PR DESCRIPTION
**Pull request by tcharding**
The HTLC expiry time requirements for rfc003 are not fully explained.

Add a section titled 'Expiry Time Considerations' that makes an attempt at
better explaining the expiry time requirements for atomicity to be guaranteed
when using rfc003.

### CODE REVIEW

**Comment by tcharding** [comment]
Is there a better way to reference a later section in an RFC document than what this patch does?

File: RFC-003-SWAP-Basic.md [hunk 1]
```diff
@@ -183,6 +185,43 @@ When Bob learns the secret from Alice's redeem activation of β-HTLC he MUST act
 He MUST make sure he does this before `alpha_expiry` or risks both losing **B** and not gaining **A**.
 To activate the redeem path he uses the secret and the procedure defined in the specification of the α-HTLC.
 
+## Expiry Time Considerations
+
+Conceptually there is a time window from when the HTLC is deployed until the HTLC expires and can be refunded.
+The protocol depends on the alpha time window being a superset of the beta time window.
+
+
```
Comment (da-kami) [inline RFC-003-SWAP-Basic.md]: 📣   I think you want to make this code-block. Please add backticks ``` before and after.

File: RFC-003-SWAP-Basic.md [hunk 2]
```diff
@@ -183,6 +185,43 @@ When Bob learns the secret from Alice's redeem activation of β-HTLC he MUST act
...
+             deploy              expiry
+
+In addition, in order for the swap to be atomic the alpha asset must be redeemed *before* `beta_expiry`.
+To be more precise; the alpha redeem transaction must have been accepted into the alpha ledger before `beta_expiry`.
+This means that the window of time for which atomicity is guaranteed for Alice is actually *smaller* than it at first appears.
+There exists a point on the beta time window depicted above (and again below) after which a redeem transaction may not get included into the ledger before the expiry time, at which time it is possible for Bob to attempt a refund transaction.
+This point is marked below with `???` because it is dependant on the ledger.
+
+
```
Comment (da-kami) [inline RFC-003-SWAP-Basic.md]: 📣   I think you want to make this code-block. Please add backticks ``` before and after.

File: RFC-003-SWAP-Basic.md [hunk 3]
```diff
@@ -183,6 +185,43 @@ When Bob learns the secret from Alice's redeem activation of β-HTLC he MUST act
...
+     deploy                                 expiry
+
+Alpha  |----------------------------------------|
+
+Beta            |--------------------|
+
+             deploy              expiry
+
+In addition, in order for the swap to be atomic the alpha asset must be redeemed *before* `beta_expiry`.
```
Comment (da-kami) [inline RFC-003-SWAP-Basic.md]:
```suggestion
In order for the swap to be atomic the alpha asset must be redeemed *before* `beta_expiry`.
```

File: RFC-003-SWAP-Basic.md [hunk 4]
```diff
@@ -183,6 +185,43 @@ When Bob learns the secret from Alice's redeem activation of β-HTLC he MUST act
...
+To be more precise; the alpha redeem transaction must have been accepted into the alpha ledger before `beta_expiry`.
+This means that the window of time for which atomicity is guaranteed for Alice is actually *smaller* than it at first appears.
+There exists a point on the beta time window depicted above (and again below) after which a redeem transaction may not get included into the ledger before the expiry time, at which time it is possible for Bob to attempt a refund transaction.
```
Comment (da-kami) [inline RFC-003-SWAP-Basic.md]:
```suggestion
There exists a point on the beta time window depicted above (and again below) after which a redeem transaction by Alice may not get included into the ledger before the expiry time, at which time it is possible for Bob to attempt a refund transaction.
```

File: RFC-003-SWAP-Basic.md (hunk 2 above)
Comment (da-kami) [inline RFC-003-SWAP-Basic.md]: 🙃  For me this second depiction was more confusing than helping me understand the text above. I'd suggest removing it for now (including the last sentence above).

File: RFC-003-SWAP-Basic.md (hunk 1 above)
Comment (tcharding) [reply RFC-003-SWAP-Basic.md]: Awesome, thanks.  Implemented as suggested.

File: RFC-003-SWAP-Basic.md (hunk 2 above)
Comment (tcharding) [reply RFC-003-SWAP-Basic.md]: Thanks, removed.

Comment (tcharding) [reply RFC-003-SWAP-Basic.md]: Thanks, removed.

File: RFC-003-SWAP-Basic.md [hunk 5]
```diff
@@ -183,6 +185,28 @@ When Bob learns the secret from Alice's redeem activation of β-HTLC he MUST act
 He MUST make sure he does this before `alpha_expiry` or risks both losing **B** and not gaining **A**.
 To activate the redeem path he uses the secret and the procedure defined in the specification of the α-HTLC.
 
+## Expiry Time Considerations
+
+Conceptually there is a time window from when the HTLC is deployed until the HTLC expires and can be refunded.
+The protocol depends on the alpha time window being a superset of the beta time window.
+
+```
+  time ->
```
Comment (da-kami) [inline RFC-003-SWAP-Basic.md]:
```suggestion
time ->
```
Looks a little weird with the indent. The line could also be removed, to me it is clear that this is related to time.

**Pull request merged**